lib/Cloudinary.pm

```diff
@@ -134,7 +134,9 @@ sub _api_sign_request {
   my @query;
 
   for my $k (@SIGNATURE_KEYS) {
-    push @query, "$k=" . url_escape($args->{$k}, '^A-Za-z0-9\-._~\/') if defined $args->{$k};
+    next unless defined $args->{$k};
+    my $v = $k eq 'public_id' ? url_escape($args->{$k}, '^A-Za-z0-9\-._~/') : $args->{$k};
```

Comment:
What would be the downside of running `url_escape` on all the values and just fixing the regex?

Reply:
As I understand, only the `public_id` parameter should be escaped when building the signature string. The API documentation says

> Create a string with the parameters used in the POST request to Cloudinary

In the documentation, it states that the `public_id` parameter should be URL-safe. It does not say this about any of the other parameters. This is probably the cause of issue https://github.com/jhthorsen/cloudinary/issues/3.

Personally, I only really care about fixing the regex because that is preventing Cloundinary from being installed with recent versions of Mojolicious. I thought I would attempt to solve the other open issue at the same time.